This patch allows us to use glibc's build system to build locales
in a package separate from glibc.

  1. Use 'localedef' from $PATH since we are not rebuilding it.
  2. Use '--no-archive' to avoid building the big locale archive, and
     because the already-built 'localedef' would want to write it
     to '/run/current-system/locale', which is not possible.
  3. Pass $(inst_complocaledir)/$$locale to install files in the right
     place, and because otherwise, 'localedef' fails with:
     "cannot write output files to `(null)'".

diff -ru glibc-2.29-old/localedata/Makefile glibc-2.29-new/localedata/Makefile
--- glibc-2.29-old/localedata/Makefile	2019-01-31 11:45:36.000000000 -0500
+++ glibc-2.29-new/localedata/Makefile	2019-07-03 22:48:35.662015071 -0400
@@ -385,7 +385,7 @@
 
 # Sometimes the whole collection of locale files should be installed.
 LOCALEDEF=I18NPATH=. GCONV_PATH=$(common-objpfx)iconvdata LC_ALL=C \
-$(rtld-prefix) $(common-objpfx)locale/localedef
+  localedef --no-archive
 install-locales: install-locale-archive
 
 # Create and install the locale-archive file.
@@ -414,7 +414,7 @@
 	input=`echo $$locale | sed 's/\([^.]*\)[^@]*\(.*\)/\1\2/'`; \
 	$(LOCALEDEF) $$flags --alias-file=../intl/locale.alias \
 		     -i locales/$$input -f charmaps/$$charset \
-		     $(addprefix --prefix=,$(install_root)) $$locale \
+		     $(addprefix --prefix=,$(install_root)) $(inst_complocaledir)/$$locale \
 	&& echo ' done';
 endef
 
